### PR TITLE
rbd: support size suffixes for size-based options

### PR DESCRIPTION
**Pull request by vumrao**
This patch will provide support for size suffixes for
below given options:

--stripe-unit [size in B/K/M]
--io-size [size in B/K/M/G/T]
--io-total [size in B/K/M/G/T]

Signed-off-by: Vikhyat Umrao <vumrao@redhat.com>

### CODE REVIEW

**Comment by vumrao** [comment]
http://tracker.ceph.com/issues/12111

**Comment by vumrao** [comment]
@jdurgin , I have checked strict_sistrtoll function:
https://github.com/ceph/ceph/blob/master/src/common/strtol.cc#L132

and it returns only value of option if user will not specify size suffix with the value which will be bytes only as  these options default value is bytes so we don't need to handle any specific case  for these options.

File: src/test/cli-integration/rbd/defaults.t [hunk 1]
```diff
@@ -94,6 +94,25 @@ Plain create with various options specified via usual cli arguments
...
+      "format": 2, 
+      "name": "test", 
+      "object_size": 4194304, 
+      "objects": 1, 
+      "order": 22, 
+      "size": 1048576, 
+      "stripe_count": 8, 
+      "stripe_unit": 1048576
+  }
+  $ rbd rm test --no-progress
```
Comment (liewegas) [inline src/test/cli-integration/rbd/defaults.t]: Can you include a test case with non-byte units too?

Comment (vumrao) [inline src/test/cli-integration/rbd/defaults.t]: Thanks Sage for review , sure I will add it!

**Comment by dachary** [comment]
Please ignore the bot failure on Ubuntu, it is a false negative that has been fixed.

**Comment by vumrao** [comment]
@dachary thanks !

**Comment by vumrao** [comment]
@jdurgin  Thanks Josh !

**Pull request merged**